include/lmms_math.h

```diff
@@ -81,26 +82,53 @@ inline auto absFraction(std::floating_point auto x) noexcept
 	return x - std::floor(x);
 }
 
-inline auto fastRand() noexcept
+
+//! @brief Returns a psuedorandom integer between 0 and 32767, inclusive.
+inline int fastRand() noexcept
 {
-	static unsigned long next = 1;
-	next = next * 1103515245 + 12345;
-	return next / 65536 % 32768;
+	thread_local unsigned long s_next = 1;
+	s_next = s_next * 1103515245 + 12345;
+	return s_next / 65536 % 32768;
 }
 
-template<std::floating_point T>
-inline auto fastRand(T range) noexcept
+
+//! @brief Returns a psuedorandom number between 0 and @p range, inclusive.
+template<typename T> requires std::is_floating_point_v<T>
+inline T fastRand(T range) noexcept
 {
-	constexpr T FAST_RAND_RATIO = static_cast<T>(1.0 / 32767);
+	constexpr auto FAST_RAND_RATIO = static_cast<T>(1.0 / 32767);
 	return fastRand() * range * FAST_RAND_RATIO;
 }
 
-template<std::floating_point T>
-inline auto fastRand(T from, T to) noexcept
+
+//! @brief Returns a psuedorandom integer between 0 and @p range, inclusive.
+template<typename T> requires std::is_integral_v<T>
+inline T fastRand(T range) noexcept
 {
-	return from + fastRand(to - from);
+	// The integer specialization of this function is kind of weird, but
+	// it is necessary to prevent massive bias away from the maximum
+	// value. FAST_RAND_RATIO here is 1 greater than normal, so it will
+	// actually result in an open-end range.
+	constexpr float FAST_RAND_RATIO = 1.f / 32768;
+	// Since it's open-end using the above ratio, increase the magnitude
+	// by 1. All values greater than range get, rounded to range, and
+	// the bias is removed.
+	const float frange = static_cast<float>(range); // Even on -O3 it casts twice without this for some reason??
+	const float r = frange + std::copysign(1.f, frange);
+	// Always round towards 0 (implicit truncation occurs during static_cast).
+	return static_cast<T>(fastRand() * r * FAST_RAND_RATIO);
```

Comment:
```suggestion
//! @brief Returns a pseudorandom integer in the range [0, @p upper)
template<std::integral T>
inline T fastRand(T upper) noexcept
{
	// The integer specialization of this function is kind of weird, but
	// it is necessary to prevent massive bias away from the maximum
	// value. FAST_RAND_RATIO here is 1 greater than normal, so it will
	// actually result in an open-ended range.
	constexpr float FAST_RAND_RATIO = 1.f / 32768;
	// Since it's open-ended using the above ratio, increase the magnitude
	// by 1. All values greater than upper get rounded to upper, and
	// the bias is removed.
	const float fupper = static_cast<float>(upper); // Even on -O3 it casts twice without this for some reason??
	const float r = fupper + std::copysign(1.f, fupper);
	// Always round towards 0 (implicit truncation occurs during static_cast).
	return static_cast<T>(fastRand() * r * FAST_RAND_RATIO);
```
Some improvements + typo fixes